### PR TITLE
Strict mode additions

### PR DESCRIPTION
**Pull request by qbob1**
Implemented: 
"noImplicitAny": true,
"noImplicitReturns": true,
"noImplicitThis": true,
"noFallthroughCasesInSwitch": true,

### CODE REVIEW

**Comment by escalonn** [comment]
you forgot `strictNullChecks` and also [`strictTemplates`](https://v9.angular.io/guide/template-typecheck#strict-mode)

File: angular/src/app/modules/account/account/account.component.html [hunk 1]
```diff
@@ -24,7 +24,9 @@ <h2 class="title has-text-light is-4">payments</h2>
             <div class="tile is-parent">
               <div class="tile box is-child">
                 <h2 class="title is-4">profiles</h2>
-                <uic-profile [profiles]="profiles$ | async"></uic-profile>
+                <div *ngIf="profiles$ | async as profiles">
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.html:27]: no need for this extra div, just put the ngif on the uic-profile like you did for the other similar cases

File: angular/src/app/modules/account/account-booking/account-booking.component.spec.ts [hunk 2]
```diff
@@ -33,7 +58,6 @@ describe('AccountBookingComponent', () => {
     component.booking = booking;
     fixture.detectChanges();
   });
-
```
Comment (escalonn) [inline angular/src/app/modules/account/account-booking/account-booking.component.spec.ts:36]: no need to remove the two newlines here & above

File: angular/src/app/modules/account/account/account.component.ts [hunk 3]
```diff
@@ -36,8 +36,8 @@ export class AccountComponent implements OnInit {
           id: '100',
           checkIn: new Date(2020, 10, 1),
           checkOut: new Date(2020, 10, 30),
-          dateCreated: null,
-          dateModified: null,
+          dateCreated: new Date(2020, 10, 1), // change to dummy data
+          dateModified: new Date(2020, 10, 1), // change to dummy data
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.ts:40]: these comments not needed here or below, it raises more questions than it answers. if you want to comment on a _change_ the commit message or the PR description is usually a better place than the code.

File: angular/src/app/modules/account/account/account.component.ts [hunk 4]
```diff
@@ -75,8 +75,8 @@ export class AccountComponent implements OnInit {
         rating: 8.5,
       },
     ]);
-    this.address$ = this.account$.pipe(map((account) => account.address));
-    this.payments$ = this.account$.pipe(map((account) => account.payments));
-    this.profiles$ = this.account$.pipe(map((account) => account.profiles));
+    this.address$ =  this.account$.pipe(map((account) => account.address   ));
+    this.payments$ = this.account$.pipe(map((account) => account.payments  ));
+    this.profiles$ = this.account$.pipe(map((account) => account.profiles  ));
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.ts:80]: 👎

File: angular/src/app/modules/account/profile/profile.component.html [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <div class="level">
   <div class="level-left">
-    <div class="level-item has-text-centered" *ngFor="let profile of profiles">
+    <div *ngIf="profiles">
+      <div class="level-item has-text-centered" *ngFor="let profile of profiles">
```
Comment (escalonn) [inline angular/src/app/modules/account/profile/profile.component.html:4]: 1. you've left out the closing tag
2. the indentation isn't right anymore
3. but just put the ngif on the existing div. (if you really need to? i thought ngfor was OK with null/undefined.)

File: angular/src/app/services/account/account.service.spec.ts [hunk 6]
```diff
@@ -14,21 +14,61 @@ import { Config } from '../../data/config.model';
...
     get() {
       const config: Config = {
         api: {
           account: 'test',
-          booking: null,
-          lodging: null,
-          monitoring: null,
+          booking: 'null',
+          lodging: 'null',
+          monitoring: 'null',
```
Comment (escalonn) [inline angular/src/app/services/account/account.service.spec.ts:54]: `'null'` is confusing

File: angular/src/app/services/account/account.service.spec.ts [hunk 7]
```diff
@@ -14,21 +14,61 @@ import { Config } from '../../data/config.model';
...
+          footer: [
+            {
+            icon: 'string',
+            text: 'string',
+            url:  'string',
+          }
+          ],
+          header: [
+            {
+            icon: 'string',
```
Comment (escalonn) [inline angular/src/app/services/account/account.service.spec.ts:66]: indentation. use prettier

File: angular/src/app/services/config/config.service.spec.ts [hunk 8]
```diff
@@ -11,12 +11,27 @@ import { HttpClient } from '@angular/common/http';
...
   const configMock: Config = {
     api: {
-      account: null,
-      booking: null,
-      lodging: null,
-      monitoring: null,
+      account: 'test',
+      booking: 'null',
+      lodging: 'null',
+      monitoring: 'null',
```
Comment (escalonn) [inline angular/src/app/services/config/config.service.spec.ts:17]: again `'null'` is confusing

File: angular/src/app/services/lodging/lodging.service.spec.ts [hunk 9]
```diff
@@ -26,12 +39,27 @@ describe('LodgingService', () => {
     get() {
       const config: Config = {
         api: {
-          account: null,
-          booking: null,
+          account: 'null',
+          booking: 'null',
           lodging: 'test',
-          monitoring: null,
+          monitoring: 'null',
```
Comment (escalonn) [inline angular/src/app/services/lodging/lodging.service.spec.ts:45]: `'null'`

File: angular/src/app/services/monitoring/monitoring.service.spec.ts [hunk 10]
```diff
@@ -11,14 +11,28 @@ describe('MonitorService', () => {
...
       const config: Config = {
         api: {
-          account: null,
-          booking: null,
-          lodging: null,
-          monitoring: null,
+          account: 'null',
+          booking: 'null',
+          lodging: 'null',
+          monitoring: 'https://ec61288c429544a6ad850b94de25004a@o388320.ingest.sentry.io/5229046',
```
Comment (escalonn) [inline angular/src/app/services/monitoring/monitoring.service.spec.ts:17]: `'null'` and also don't put the real monitoring url in the unit tests

File: angular/tsconfig.json [hunk 11]
```diff
@@ -8,14 +8,19 @@
...
     "resolveJsonModule": true,
     "sourceMap": true,
+    "strictNullChecks": true,
     "target": "es2015",
-    "lib": ["es2018", "dom"]
+    "lib": ["es2018", "dom"],
   },
   "angularCompilerOptions": {
-    "fullTemplateTypeCheck": true,
+    "strictTemplates": true,
```
Comment (escalonn) [inline angular/tsconfig.json:23]: alphabetize stricttemplates

File: angular/src/app/modules/account/account-booking/account-booking.component.spec.ts (hunk 2 above)
Comment (escalonn) [reply angular/src/app/modules/account/account-booking/account-booking.component.spec.ts:36]: again, restore newlines

File: angular/src/app/services/config/config.service.spec.ts (hunk 8 above)
Comment (escalonn) [reply angular/src/app/services/config/config.service.spec.ts:17]: `'null'` still here

File: angular/src/app/modules/account/account-booking/account-booking.component.spec.ts (hunk 2 above)
Comment (qbob1) [reply angular/src/app/modules/account/account-booking/account-booking.component.spec.ts:36]: for some reason this was giving me linting issues last night 🤷

File: angular/src/app/modules/account/profile/profile.component.html (hunk 5 above)
Comment (escalonn) [reply angular/src/app/modules/account/profile/profile.component.html:4]: remark #3 unaddressed, an extra div isnt motivated for this PR

File: angular/src/app/modules/lodging/lodging/lodging.component.html [hunk 12]
```diff
@@ -123,7 +123,8 @@ <h2 class="title is-4">rooms</h2>
               <div class="level">
                 <div class="level-item">
                   <figure class="image is-64x64">
-                    <img alt="placeholder image" class="is-rounded" src="https://bulma.io/images/placeholders/128x128.png">
+                    <img alt="placeholder image" class="is-rounded"
+                      src="https://bulma.io/images/placeholders/128x128.png">
```
Comment (escalonn) [inline angular/src/app/modules/lodging/lodging/lodging.component.html:127]: the diff to this file is unnecessary in this PR

**Pull request merged**